cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/distributed/k8s/SparkContainerDriverLauncher.java

```diff
@@ -125,8 +135,50 @@ public static void main(String[] args) throws Exception {
     CConfiguration cConf = CConfiguration.create(new File(CCONF_PATH));
     Configuration hConf = new Configuration();
     hConf.addResource(new org.apache.hadoop.fs.Path("file:" + new File(HCONF_PATH).getAbsolutePath()));
-    ArtifactLocalizerClient fetchArtifacts = createArtifactLocalizerClient(cConf, hConf);
 
+    if (artifactFetcherUri == null) {
+      localizeArtifactsFromAppfabric(cConf, hConf);
+    } else {
+      try {
+        localizeArtifactsBundle(artifactFetcherUri);
+      } catch (Exception e) {
+        localizeArtifactsFromAppfabric(cConf, hConf);
```

Comment:
Is there a way to add log line or something in case if we read from appfabric? Otherwise it will be hard to know where the artifacts are being query and may cause additional load on appfabric.